core/src/main/scala/io/finch/request/package.scala

```diff
@@ -49,14 +129,22 @@ package object request {
     def apply[Req](req: Req)(implicit ev: Req => HttpRequest): Future[A]
 
     def flatMap[B](fn: A => RequestReader[B]) = new RequestReader[B] {
+      val item = MultipleItems
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) = self(req) flatMap { fn(_)(req) }
     }
 
     def map[B](fn: A => B) = new RequestReader[B] {
+      val item = self.item
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) = self(req) map fn
     }
     
+    def mapFuture[B](fn: A => Future[B]) = new RequestReader[B] {
```

Comment:
I have been thinking about the name `mapFuture` again. I think it is a bad name and needs to go, in particular as it really is a `flatMapFuture` and we should not use the type name in the method name, so that we could use the same name for other embedded monads.

The names `embed` or `hoist` are nice and short, however, I feel they don't capture the similarity to `map` and `flatMap` in any way (and as a non-native speaker I did not even know what hoist means). If we could live with a longer name, I think `embedFlatMap` is a fairly precise description of what that method does. What do you think?